prompting/weight_setting/weight_setter.py

```diff
@@ -0,0 +1,185 @@
+from loguru import logger
+import bittensor as bt
+import numpy as np
+import os
+import pandas as pd
+
+from prompting import __spec_version__
+from prompting.settings import settings
+from prompting.utils.uids import get_uids
+from prompting.utils.misc import ttl_get_block
+from prompting.base.loop_runner import AsyncLoopRunner
+from prompting import mutable_globals
+from prompting.rewards.reward import WeightedRewardEvent
+from prompting.tasks.task_registry import TaskRegistry, TaskConfig
+from prompting.tasks.inference import InferenceTask
+from prompting.llms.model_zoo import ModelZoo
+
+PAST_WEIGHTS: list[np.ndarray] = []
```

Comment:
Is there a specific reason for a mix of OOP and procedural style? Can't we move everything to `WeightSetter`?

Reply:
We could do - I just normally prefer to have constant defined at the top of the file so they're easier to spot/edit quickly